### PR TITLE
Added option to ignore file paths

### PR DESCRIPTION
**Pull request by ShaharNaveh**
This is needed both for #32942 and to #33394 to ignore the ```pandas/tests``` directory

### CODE REVIEW

**Comment by alimcmaster1** [comment]
If you merge master test failures will be resolved

File: scripts/validate_unwanted_patterns.py [hunk 1]
```diff
@@ -337,17 +336,23 @@ def main(
...
+
         for file_name in files:
             if not any(
                 file_name.endswith(extension) for extension in FILE_EXTENSIONS_TO_CHECK
             ):
                 continue
 
             file_path = os.path.join(subdir, file_name)
-            with open(file_path, "r") as file_obj:
+            with open(file_path, mode="r") as file_obj:
```
Comment (jbrockmendel) [inline scripts/validate_unwanted_patterns.py:355]: is making "mode" explicit necesary?  its not a pattern we used elsewhere

File: scripts/validate_unwanted_patterns.py [hunk 2]
```diff
@@ -389,6 +394,11 @@ def main(
         default="py,pyx,pxd,pxi",
         help="Coma seperated file extensions to check.",
     )
+    parser.add_argument(
+        "--excluded-file-paths",
+        default="asv_bench/env",
+        help="Coma seperated file extensions to check.",
```
Comment (jbrockmendel) [inline scripts/validate_unwanted_patterns.py:400]: Coma -> Comma (i hope) 
seperated -> separated

looks like both of these need fixing on L395 too

File: scripts/validate_unwanted_patterns.py [hunk 3]
```diff
@@ -337,17 +336,23 @@ def main(
                     )
                 )
 
+    FILE_EXTENSIONS_TO_CHECK: FrozenSet[str] = frozenset(
```
Comment (WillAyd) [inline scripts/validate_unwanted_patterns.py:339]:
```suggestion
    FILE_EXTENSIONS_TO_CHECK = frozenset(
```

Mypy will infer this from the right hand side

File: scripts/validate_unwanted_patterns.py [hunk 4]
```diff
@@ -337,17 +336,23 @@ def main(
...
+        file_extensions_to_check.split(",")
+    )
+    PATHS_TO_IGNORE: FrozenSet[str] = frozenset(excluded_file_paths.split(","))
```
Comment (WillAyd) [inline scripts/validate_unwanted_patterns.py:342]:
```suggestion
    PATHS_TO_IGNORE = frozenset(excluded_file_paths.split(","))
```

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by MomIsBestFriend** [comment]
Sorry, I got into other obligation in the past week and a half, I'll try to finish what I started as much as I can.

**Comment by jreback** [comment]
can you merge master

**Comment by jreback** [comment]
closing as stale, if you want to continue pls ping.

**Pull request closed without merging**